import { MdxLayout, ArticleSection, ArticleNavigationContainer, MathFormula, PreviousArticleLink } from "../../../components";
import {
  FormulaReview,
  NegativeAutoregAccumulationTutorial,
  NegativeAutoregResponseTimeComparisonTutorial,
} from "../../../interactives/systems-biology";

export const metadata = {
  title: "Dynamics of Negative Autoregulation",
  description: "How self-repression speeds up the response time of gene circuits",
  keywords: "network motifs, random networks, autoregulation, gene expression, transcription factors, systems biology, cell signaling",
  ogImage: "https://i.ibb.co/Nnbfc6y/genetic-circuit.png",
  url: "https://newtinteractive.com/series/systems-biology/autoregulation-2",
  series: {
    name: "Systems Biology",
    href: "/series/systems-biology",
  },
};

# Dynamics of Negative Autoregulation

## How self-repression speeds up the response time of gene circuits

In the previous article, we learned that autoregulation, and in particular negative autoregulation, is a motif in biological transcriptional networks; it occurs far more than it would at random. But what, exactly, is negative autoregulation? How does it work, and how does it provide an advantage for organisms?

Before we answer these questions, let's review some of the key concepts and equations we've learned so far. If you're already comfortable with them, feel free to skip ahead.

<ArticleSection variant="small">
  <FormulaReview />
</ArticleSection>

---

Negative autoregulation occurs when a transcription factor represses its own expression; the protein that the gene expresses binds to its own promoter and inhibits the production of mRNA. As the protein concentration increases, it ends up reducing its own rate of production.

In [Transcription Network Basics: Part Three](/series/systems-biology/transcription-network-basics-3), we set up a baseline for dynamics and response time for simple regulation, and identified it as an important feature of cellular performance. Let's now look at the dynamics and response time for negative autoregulation and compare.

To keep things simple, let's begin with using a logic approximation:

<ArticleSection variant="math">
  <MathFormula tex="f(X) = \beta \cdot \theta(X < K)" />
</ArticleSection>

so the production rate is either 0, when <MathFormula tex="X" /> is less than the threshold <MathFormula tex="K" />, or at its maximum <MathFormula tex="\beta" />, when <MathFormula tex="X" /> is greater than <MathFormula tex="K" />.

To analyze its response time, let's consider the scenario where the protein concentration is initially 0. As time progresses, the unrepressed promoter produces the protein at the full-speed rate <MathFormula tex="\beta" />, and the concentration of protein changes according to the equation:

<ArticleSection variant="math">
  <MathFormula tex="\dfrac{dX}{dt} = \beta - \alpha X \qquad \text{while} \thickspace X < K" />
</ArticleSection>

At early times, since <MathFormula tex="X" /> levels are very low, the removal term is very small (<MathFormula tex="\alpha X \ll \beta" />) and thus, negligible. This leaves us with roughly a linear accumulation formula for <MathFormula tex="X" />:

<ArticleSection variant="math">
  <MathFormula tex="X(t) \approx \beta t \qquad \text{while} \thickspace X < K \enspace \text{and} \enspace X \ll \dfrac{\beta}{\alpha}" />
</ArticleSection>

Once <MathFormula tex="X" /> reaches the threshold <MathFormula tex="K" />, production immediately goes to zero (as per our step function). To make all of this a little more clear, let's look at this graphically:

<ArticleSection variant="medium">
  <NegativeAutoregAccumulationTutorial />
</ArticleSection>

From the graph, we can see that the steady state concentration of <MathFormula tex="X" /> is the same as its repression coefficient:

<ArticleSection variant="math">
  <MathFormula tex="X_{st} = K" />
</ArticleSection>

### Response time

Remember, from [Transcription Network Basics: Part Three](/series/systems-biology/transcription-network-basics-3), that response time <MathFormula tex="T_{1/2}" /> is defined as the time it takes a protein to get to half of its steady-state level. Sticking with our simplified linear accumulation formula, that means:

<ArticleSection variant="math">
  <MathFormula tex="X(T_{1/2_{NAR}}) = \beta \cdot T_{1/2_{NAR}} = \dfrac{X_{st}}{2} = \dfrac{K}{2}" />
</ArticleSection>

and so:

<ArticleSection variant="math">
  <MathFormula tex="T_{1/2_{NAR}} = \dfrac{K}{2\beta}" />
</ArticleSection>

From this, we can see that the stronger the maximal promoter activity <MathFormula tex="\beta" />, the faster the response time.

Now, how does this compare with the response time for simple regulation? Before we jump into that, we need to ensure that the comparison is fair. This is done with something known as a **mathematically controlled comparison**; essentially, both designs should have the same target while keeping as many biochemical parameters identical as possible. In this comparison, the target is to reach the same steady-state level, a crucial quantity for optimal cellular performance. 

Comparing the steady state formulas for both designs:

<ArticleSection variant="math">
  <MathFormula tex="X_{st_{simple}} = \dfrac{\beta_{simple}}{\alpha_{simple}}" />
  <MathFormula className="mt-6" tex="X_{st_{NAR}} = K" />
</ArticleSection>

which, when equated, gives us:

<ArticleSection variant="math">
  <MathFormula tex="K = \dfrac{\beta_{simple}}{\alpha_{simple}}" />
</ArticleSection>

As you'll soon see in the interactive chart below, keeping the same steady state levels mean providing a stronger promoter for the negative autoregulation design, to compensate for its self-repression. 

<ArticleSection variant="large">
  <NegativeAutoregResponseTimeComparisonTutorial />
</ArticleSection>


<ArticleSection variant="medium" className="mb-0 md:mb-0" >
  <ArticleNavigationContainer>
    <PreviousArticleLink
      href="/series/systems-biology/autoregulation-1"
      title="Autoregulation as a Network Motif"
    />
  </ArticleNavigationContainer>
</ArticleSection>


export default function MDXPage({ children }) {
  return <MdxLayout metadata={metadata}>{children}</MdxLayout>;
}
