import { MdxLayout, ArticleSection, MathFormula, ImageSeries, Quote, UnorderedList, TippyTooltip, TippyTooltipContent, PostArticleSubscribe, Sheet, SheetTrigger, SheetContent, SheetTitle, PreviousArticleLink, ArticleNavigationContainer } from "../../../components";
import { VisuallyHidden } from "@radix-ui/react-visually-hidden";
import { AutoregulationNetworkComparisonTutorial } from "../../../interactives/systems-biology";
import ErdosRenyiGNMNetworkWithProvider from "../../../interactives/erdos-renyi-graph";
import * as Accordion from "@radix-ui/react-accordion";
import { FiInfo, FiChevronDown } from "react-icons/fi";

export const metadata = {
  title: "Autoregulation: Part One",
  description: "Explore the concepts of network motifs, random networks and autoregulation in systems biology.",
  keywords: "network motifs, random networks, autoregulation, gene expression, transcription factors, systems biology, cell signaling",
  ogImage: "https://i.ibb.co/Nnbfc6y/genetic-circuit.png",
  url: "https://newtinteractive.com/series/systems-biology/autoregulation-1",
};

export function CalculationTooltip() {
  return (
    <TippyTooltip
      highlightColor="indigo-50"
      iconColor="indigo-100"
      content={<TippyTooltipContent>I replaced the scientific notation numbers (<MathFormula variant="popover" tex="10^{10}" />) with words (10 billion) for ease of reading.</TippyTooltipContent>}
    >
      As Uri Alon calculates in <i>An Introduction to Systems Biology</i>
    </TippyTooltip>
  )
}

export function StandardDeviationSheet() {
  return (
    <Sheet>
      <SheetTrigger asChild>
        <span className="underline decoration-indigo-500 decoration-2 underline-offset-[3px] hover:bg-indigo-100 cursor-pointer">
          approximately the square root of the mean
        </span>
      </SheetTrigger>
      <SheetContent>
        <VisuallyHidden>
          <SheetTitle>Standard Deviation in Random Networks</SheetTitle>
        </VisuallyHidden>
        <div className="flex flex-col text-md text-sm">
          <span className="mb-3 mt-3">
            this comes from the binomial distribution. Whether an edge is drawn in the network is like a yes-no question, where the probability of yes is <MathFormula variant="popover" tex="p" /> and the probability of no is <MathFormula variant="popover" tex="q = 1 - p" />.
          </span>
          <span className="mb-3">
            In such cases, with <MathFormula variant="popover" tex="n" /> trials, the mean is:
          </span>
          <MathFormula className="ml-6" variant="popover" tex="\mu = n \times p" />
          <span className="mt-6 mb-3">
            which is the same as the <MathFormula variant="popover" tex="\lang N_{self} \rang_{rand}" /> formula we have. The standard deviation is:
          </span>
          <MathFormula className="ml-6" variant="popover" tex="\sigma = \sqrt{n \times p \times q}" />
          <MathFormula className="ml-6" variant="popover" tex="\sigma = \sqrt{n \times p \times (1 - p)}" />
          <span className="mt-6 mb-3">
            in our example, probability is <MathFormula variant="popover" tex="1/n" />, and number of trials (number of connections) is <MathFormula variant="popover" tex="M" />, so:
          </span>
          <MathFormula className="ml-6" variant="popover" tex="\sigma = \sqrt{M \times \dfrac{1}{n} \times \left(1 - \dfrac{1}{n}\right)}" />
          <MathFormula className="ml-6" variant="popover" tex="\sigma = \sqrt{\dfrac{M}{n} \times \left(1 - \dfrac{1}{n}\right)}" />
          <span className="mt-6 mb-3">
            for large <MathFormula variant="popover" tex="n" />, <MathFormula variant="popover" tex="\left(1 - \dfrac{1}{n}\right)" /> is approximately 1, so that gives:
          </span>
          <MathFormula className="ml-6" variant="popover" tex="\sigma \approx \sqrt{\dfrac{M}{n}}" />
        </div>
      </SheetContent>
    </Sheet>
  );
}

export function ErdosReyniAccordion() {
  return (
    <Accordion.Root type="single" collapsible className="w-full bg-slate-50 rounded-md max-w-prose self-center px-5 md:px-6 py-4 md:py-5">
      <Accordion.Item className="text-sm md:text-base font-body" value="item-1">
        <Accordion.Trigger className="flex items-center text-base">
          <FiInfo className="mr-1.5" /> A Note on calculating Erdős-Rényi Networks <FiChevronDown className="ml-1" />
        </Accordion.Trigger>
        <Accordion.Content className="mt-4 py-3 border-t border-slate-200">
          <span className="text-base text-slate-800 mb-3">The ER model comes in two variants, each with a different method of generating the random network:</span>

          <UnorderedList className="text-base text-slate-800 mt-3 mb-0">
            <li className="text-base">
              <MathFormula variant="tutorial" tex="G(n, p)" />: start with <MathFormula variant="tutorial" tex="n" /> nodes and then connect each pair of nodes (create an edge between them) with a probability <MathFormula variant="tutorial" tex="p" />, independent of all other pairs.
            </li>
            <li className="text-base">
              <MathFormula variant="tutorial" tex="G(n, M)" />: start with <MathFormula variant="tutorial" tex="n" /> nodes and randomly select <MathFormula variant="tutorial" tex="M" /> edges (from all possible edges) to connect the nodes. Useful for generating graphs with a specific number of edges.
            </li>
          </UnorderedList>
          <span className="text-base text-slate-800 mt-3">
            The interactive above uses the <MathFormula variant="tutorial" tex="G(n, M)" /> method. The math that follows works for both.
          </span>
        </Accordion.Content>
      </Accordion.Item>
    </Accordion.Root>
  )
}

# Autoregulation as a Network Motif

## Random networks, finding patterns, and genes that regulate their own expression

In the previous section, we looked at a single interaction in a transcription network X → Y, what it represents, and its dynamics. Now, let's look at an actual transcription network &mdash; the same one we introduced at the end of Part One: a network of about 20% of *E.Coli*'s genes.

<ArticleSection variant="medium" className="mt-5 md:mt-4">
  <ImageSeries
    images={[
      {
        src: "/images/transcription-network-with-autoregulation.png",
        alt: "Complex transcription network diagram of E. coli showing interconnected genes and regulatory elements",
        width: 500,
        height: 500,
        caption: <span>20% of <i>E.Coli</i>'s transcription network, with the transcription factors split into ones with and without autoregulation. Taken from <i>An Introduction to Systems Biology</i> (Uri Alon, 2006)</span>,
      }
    ]}
  />
</ArticleSection>

Even with just 20% of the genes, the network looks pretty complex. In order to understand it, let's break it down into smaller networks and look for patterns &mdash; patterns that will hopefully help us understand the dynamics of the whole network.

Before we jump into breaking down this graph though, it's important to understand some context behind it.

The graph above is a snapshot in time. But cells, particularly bacterial ones, are a dynamic place. They divide. And mutate. In the right conditions, relentlessly and endlessly. <CalculationTooltip />

> A single bacterium placed in a test tube with 10 mL of liquid nutrient grows and divides to reach a saturating population of about 10 billion cells within less than a day. This population therefore underwent 10 billion DNA replications. Since the mutation rate is about one in a million per letter per replication, the population will include, for each letter in the genome, 10 different bacteria with a mutation in that letter. Thus, a change of any DNA letter can be rapidly reached in bacterial populations.

<div className="mt-8" />
A single letter mutation in the DNA sequence of a promoter can prevent the usual binding of a transcription factor, and thus erase an arrow from the network. Similarly, mutations in the binding site, repositioning of DNA segments, or insertions of DNA from other sources (genetic recombination) can create new binding sites for transcription factors, and so, add arrows to the network.

Arrows, then, can be lost or added pretty easily and quite often. This means that pathways that are preserved across evolutionary timescales have to constantly be selected for against the random forces of mutations. In other words, if a pattern does emerge much more often than it would at random, then it must provide some evolutionary advantage to the organism. The loss of a pathway that's crucial for glucose metabolism, for example, would be detrimental to the cell's survival, so we're likely to see that pathway preserved in the network.

<div className="mt-8" />
### Random Networks and Network Motifs

Patterns that occur in a real network significantly more than they would at random are called **network motifs**. In order to determine this statistical significance, the real network needs to be compared to a random network; i.e. a network with the same parameters (number of nodes and arrows), but where the connections are assigned randomly.

There are a number of random network models, but we'll use the simplest one: the Erdős-Rényi (ER) model. The way it works is fairly simple: for every pair of nodes, the decision to connect them or not is made at random.

As an example, below is a comparison between a 'real' network and a random network for 14 nodes and 20 arrows (in mathematics, these 'arrows', or connections or pathways, are called edges). Try regenerating the random network to see different possibilities. Take note of the number of self-arrows &mdash; arrows that connect a node to itself &mdash; in the random one. How many are usually there, and how does it compare to the 'real' network? Later, we'll use the incidence of self-arrows as a starting point to compare networks and determine motifs.

<ArticleSection variant="large" className="mt-9 md:mt-8">
  <AutoregulationNetworkComparisonTutorial />
</ArticleSection>

<ArticleSection variant="medium" className="mt-0 md:mt-0">
  <ErdosReyniAccordion />
</ArticleSection>


### Autoregulation

The self-arrows in the network denote a gene that regulates its own expression, a process known as **autoregulation**. In the *E.coli* transcription network image (reproduced below), the black circles represent transcription factor genes with autoregulation, and the grey circles represent transcription without.

<ArticleSection variant="medium" className="mt-5 md:mt-4">
  <ImageSeries
    images={[
      {
        src: "/images/transcription-network-with-autoregulation.png",
        alt: "Complex transcription network diagram of E. coli showing interconnected genes and regulatory elements",
        width: 500,
        height: 500,
        caption: <span>20% of <i>E.Coli</i>'s transcription network, with the transcription factors split into ones with and without autoregulation. Taken from <i>An Introduction to Systems Biology</i> (Uri Alon, 2006)</span>
      }
    ]}
  />
</ArticleSection>

There are 40 autoregulatory transcription factors in this network. Thirty-four of these are repressers &mdash; proteins that repress their own expression:  **negative autoregulation**.

Does autoregulation occur in this network significantly more often than it would at random? To answer that question, we'll need to compare it to the average number of self-arrows that would occur in an Erdős-Rényi network of the same size.

---
<div className="mt-8" />

Let's say we have two nodes, <MathFormula variant="regular" tex="A" /> and <MathFormula variant="regular" tex="B" />, and we start on node <MathFormula variant="regular" tex="A" />. There are two possible arrows: either it points to itself, or it points to <MathFormula variant="regular" tex="B" />. If we pick at random, the probability of a self-arrow is <MathFormula variant="regular" tex="1/2" />. If there are 3 nodes, then there are three possible arrows from <MathFormula variant="regular" tex="A" />, and the probability of a self-arrow is <MathFormula variant="regular" tex="1/3" />. Following this logic, if there are <MathFormula variant="regular" tex="n" /> nodes, then the probability of a self arrow for any given node is:

<ArticleSection variant="medium" className="justify-center mx-auto mt-5 md:mt-4">
  <MathFormula variant="regular" tex="p_{self} = \dfrac{1}{n}" />
</ArticleSection>

The random network needs to have the same number of arrows as the real network for the comparison to be valid. Let's denote this number of arrows as <MathFormula variant="regular" tex="M" />. If the <MathFormula variant="regular" tex="M" /> arrows are placed randomly, then the average number of self-arrows we'll get is just the number of arrows multiplied by the probability of a self-arrow, which gives:

<ArticleSection variant="medium" className="justify-center mx-auto mt-5 md:mt-4">
  <MathFormula variant="regular" tex="\lang N_{self} \rang_{rand} = M \times p_{self} = \dfrac{M}{n}" />
</ArticleSection>

and the standard deviation is <StandardDeviationSheet />, so:

<ArticleSection variant="medium" className="justify-center mx-auto mt-5 md:mt-4">
  <MathFormula variant="regular" tex="\sigma_{N_{self}} = \sqrt{\lang N_{self} \rang_{rand}} = \sqrt{\dfrac{M}{n}}" />
</ArticleSection>

For the *E.coli* network above, the number of nodes and arrows are <MathFormula variant="regular" tex="n = 424" /> and <MathFormula variant="regular" tex="M = 519" />. Thus, the average number of self-arrows and its standard deviation are:

<ArticleSection variant="medium" className="justify-center mx-auto mt-5 md:mt-4">
  <MathFormula variant="regular" className="mb-6" tex="\lang N_{self} \rang_{rand} = \dfrac{519}{424} \approx 1.23" />
  <MathFormula variant="regular" className="self-center ml-5 md:ml-6" tex="\sigma_{N_{self}} = \sqrt{\dfrac{519}{424}} \approx 1.11" />
</ArticleSection>

A random network with 424 nodes and 519 arrows expects only a little over 1 self-arrow. The real network, remember, has 40 self-arrows, which is about 35 standard deviations away from the expected value. That's a very high statistical significance.

Negative autoregulation itself, with 34 self-arrows, is about 30 standard deviations away. It's safe to say, then, that autoregulation, and in particular, negative autoregulation, is a network motif in transcription networks. Remember, earlier, we said that a network motif implies some evolutionary advantage &mdash; what's the advantage that negative autoregulation provides?

We'll explore this question in the next article. For now, feel free to play around with an Erdős-Rényi network generator below and see how the network changes when you change some parameters. You can also check it out in the <a className="text-slate-800 hover:text-slate-900 underline underline-offset-1 decoration-slate-700" href="/blocks/erdos-renyi-graph">Erdős-Rényi Graphs block</a>.

<ArticleSection variant="large" className="mt-9 md:mt-8">
  <ErdosRenyiGNMNetworkWithProvider />
</ArticleSection>

<ArticleSection variant="medium" className="mb-0 md:mb-0" >
  <ArticleNavigationContainer>
    <PreviousArticleLink
      href="/series/systems-biology/transcription-network-basics-3"
      title="Transcription Network Basics: Part Three"
    />
  </ArticleNavigationContainer>
</ArticleSection>

<PostArticleSubscribe />

export default function MDXPage({ children }) {
  return (
    <MdxLayout metadata={metadata}>{children}</MdxLayout>
  );
}
